mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1375,7 +1454,7 @@ public IAsyncResult BeginConnect (EndPoint end_point, AsyncCallback callback, ob
 			is_bound = false;
 			connect_in_progress = true;
 
-			socket_pool_queue (SocketAsyncWorker.Dispatcher, sockares);
+			IOSelector.Add (sockares.handle, new IOSelectorJob (IOOperation.Write, s => BeginConnectCallback ((SocketAsyncResult) s), sockares));
```

Comment:
Shouldn't sockares.handle have the same value as the Socket handle itself?
If so, why do we need to store the handle in the sockares?